url587829ef81_champdavis_KhovanovHomology_tgz-pJ8-KhovanovMapJ8/src/MergeMap.java

```diff
@@ -0,0 +1,77 @@
+import java.util.*;
+
+public class MergeMap implements DMap{
+       
+    private KhovanovMapNode fromNode;
+    private KhovanovMapNode toNode;
+    private ArrayList<SeifertCircle> fromCircles;
+    private SeifertCircle toCircle;
+
+    public MergeMap(KhovanovMapNode f, KhovanovMapNode t, ArrayList<SeifertCircle> fc, SeifertCircle tc){
+        this.fromNode = f;
+        this.toNode = t;
+        this.fromCircles = fc;
+        this.toCircle = tc;  
+        
+        addTensorArrows();
+        System.out.println(toString());
+    }
+    
+    public KhovanovMapNode getToNode(){
+        return this.toNode;
+    }
+    
+    public void addTensorArrows(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.